src/utils/rocpd_data.py

```diff
@@ -0,0 +1,93 @@
+import csv
+import sqlite3
+from contextlib import closing
+
+from utils.logger import console_error
+
+# From schema definition in source/share/rocprofiler-sdk-rocpd/data_views.sql in rocprofiler-sdk repository
+COUNTERS_COLLECTION_QUERY = """
+SELECT
+    agent_id as GPU_ID,
+    dispatch_id as Dispatch_ID,
+    grid_size as Grid_Size,
+    workgroup_size as Workgroup_Size,
+    lds_block_size as LDS_Per_Workgroup,
+    scratch_size as Scratch_Per_Workitem,
+    vgpr_count as Arch_VGPR,
+    accum_vgpr_count as Accum_VGPR,
+    sgpr_count as SGPR,
+    kernel_name as Kernel_Name,
+    start as Start_Timestamp,
+    end as End_Timestamp,
+    kernel_id as Kernel_ID,
+    counter_name as Counter_Name,
+    value as Counter_Value
+FROM counters_collection
+"""
+
+
+def convert_db_to_csv(
+    db_path: str,
+    csv_file_path: str,
+) -> None:
+    """
+    Read rocpd database and write to CSV file
+    """
+    # Read counters_collection view from the database and write to CSV
+    try:
+        with closing(sqlite3.connect(db_path)) as conn:
+            with closing(conn.execute(COUNTERS_COLLECTION_QUERY)) as cursor:
+                with open(csv_file_path, "w", newline="") as csvfile:
+                    writer = csv.writer(csvfile)
+                    writer.writerow(
+                        [description[0] for description in cursor.description]
+                    )
+                    for row in cursor:
+                        writer.writerow(row)
+    except (sqlite3.DatabaseError, IOError) as e:
+        console_error(f"Error converting database to CSV: {e}")
+
+def process_rocpd_csv(df):
+    """
+    Merge counters across unique dispatches from the input dataframe and return processed dataframe.
+    """
+    # Only import pandas if needed
+    import pandas as pd
+
+    data = list()
+    # Group by unique kernel and merge into a single row
+    for _, group_df in df.groupby(
+        [
+            "Dispatch_ID",
```

Comment:
I'd suggest to put this kind of definition in a centrelized place because we are using those a few places.
In the past, rocprof change those without notification...It is even better to check the columns in advacned.

Reply:
In the past, we put comon defs in schema.py. 
Or somewhere root for profiler_base/v3.